pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -661,29 +682,14 @@ func (r *Reconciler) reconcileGenericTarget(ctx context.Context, externalSecret
 
 	case esv1.CreatePolicyMerge:
 		// for Merge policy, only update if resource exists
-		existing, getErr := r.getGenericResource(ctx, log, externalSecret)
-		if getErr != nil && !apierrors.IsNotFound(getErr) {
-			r.markAsFailed("could not get target resource", getErr, externalSecret, syncCallsError.With(resourceLabels), esv1.ConditionReasonResourceSyncedError)
-			return ctrl.Result{}, getErr
-		}
-
 		if existing == nil || existing.GetUID() == "" {
 			r.markAsDone(externalSecret, start, log, esv1.ConditionReasonResourceMissing, "resource will not be created due to CreationPolicy=Merge")
 			return r.getRequeueResult(externalSecret), nil
 		}
 
-		obj.SetResourceVersion(existing.GetResourceVersion())
-		obj.SetUID(existing.GetUID())
-
 		// update the existing resource
 		err = r.updateGenericResource(ctx, log, externalSecret, obj)
 	case esv1.CreatePolicyOrphan, esv1.CreatePolicyOwner:
-		existing, getErr := r.getGenericResource(ctx, log, externalSecret)
-		if getErr != nil && !apierrors.IsNotFound(getErr) {
-			r.markAsFailed("could not get target resource", getErr, externalSecret, syncCallsError.With(resourceLabels), esv1.ConditionReasonResourceSyncedError)
-			return ctrl.Result{}, getErr
-		}
-
 		if existing != nil {
 			obj.SetResourceVersion(existing.GetResourceVersion())
 			obj.SetUID(existing.GetUID())
```

Comment:
Why are we removing these only for CreationPolicy=Merge?

Reply:
Removing? 🤔